spinoso-string/src/lib.rs

```diff
@@ -303,17 +145,18 @@ impl String {
     /// ```
     /// use spinoso_string::{Encoding, String};
     ///
-    /// const S: String = String::new();
+    /// let S: String = String::new();
     /// assert_eq!(S.encoding(), Encoding::Utf8);
     /// ```
     ///
     /// [conventionally UTF-8]: crate::Encoding::Utf8
     #[inline]
     #[must_use]
-    pub const fn new() -> Self {
+    pub fn new() -> Self {
```

Comment:
Can no longer be `const` due to relying on the `EncodedString` constructor

Reply:
I believe the `EncodedString` constructor can be `const` too. Can we try to preserve `const` here?